kernel/tests/write.rs

```diff
@@ -791,3 +796,367 @@ async fn test_append_timestamp_ntz() -> Result<(), Box<dyn std::error::Error>> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_append_variant() -> Result<(), Box<dyn std::error::Error>> {
+    // setup tracing
+    let _ = tracing_subscriber::fmt::try_init();
+    fn unshredded_variant_schema_flipped() -> DataType {
+        DataType::variant_type([
+            StructField::not_null("value", DataType::BINARY),
+            StructField::not_null("metadata", DataType::BINARY),
+        ])
+    }
+    fn variant_arrow_type_flipped() -> ArrowDataType {
+        let metadata_field = Field::new("metadata", ArrowDataType::Binary, false);
+        let value_field = Field::new("value", ArrowDataType::Binary, false);
+        let fields = vec![value_field, metadata_field];
+        ArrowDataType::Struct(fields.into())
+    }
+
+    // create a table with VARIANT column
+    let table_schema = Arc::new(StructType::new(vec![
+        StructField::nullable("v", unshredded_variant_schema())
+            .with_metadata([("delta.columnMapping.physicalName", "col1")])
+            .add_metadata([("delta.columnMapping.id", 1)]),
```

Comment:
any reason to not just do 
```suggestion
            .with_metadata([
                ("delta.columnMapping.physicalName", "col1",
                ("delta.columnMapping.id", 1)
            ]),
```

Reply:
Yeah, so the value type needs to be the same in all elements of the array. I'm sure the high level APIs can be fixed to allow for dynamic type inference but keeping it like this is simpler for this PR.